src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -85,10 +88,41 @@ public Response forward(@NotNull @Valid RestAction ra) {
         receivedActions.increment();
 
         try {
-            restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
-        } catch (BadRequestException e) {
-            // The following line is required to forward the HTTP 400 error message.
-            return Response.status(BAD_REQUEST).entity(e.getMessage()).build();
+            final Response response = this.restValidationClient.validate(ra.getBundle(), ra.getApplication(), ra.getEventType());
+
+            // Close the response to avoid memory leaks.
+            if (response != null) {
+                response.close();
+            }
+        } catch (final WebApplicationException e) {
+            // Build a nice error message for the caller.
+            final Response response = e.getResponse();
+            final String incomingErrorMessage = response.readEntity(String.class);
+            // Close the response to avoid memory leaks.
+            response.close();
+
+            Log.debugf(
+                "Unable to validate the provided rest action due to notifications-backend responding with an unexpected error. Received status code: %s, received error message: %s, received REST action in the gateway: %s",
+                response.getStatus(),
+                incomingErrorMessage,
+                ra
+            );
+
+            // Notify the caller about the error.
+            return Response
+                .status(BAD_REQUEST)
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON)
+                .entity(buildResponseEntity(false, incomingErrorMessage))
+                .build();
```

Comment:
This is fine when `backend` returns a 400 because `incomingErrorMessage` will  contain a user-friendly message. However, I don't think we should return `BAD_REQUEST` regardless of the actual status returned by `backend`. I would map 400 from `backend` to BAD_REQUEST in `gw` and any other status (most likely a 5xx family status 99% of the time) from `backend` to SC_SERVICE_UNAVAILABLE in `gw`.